### DISCUSSION TITLE
Properly closing the client

### DISCUSSION THREAD

**Discussion by Magaud59** [MPh-py/MPh#173] [General]
Dear MPH users and developers,

I'm using COMSOL in parallel between the COMSOL GUI and the python console.
Here is my procedure:
I'm starting my COMSOL server, open my model in the COMSOL GUI and connect via the GUI to the comsol server.
I'm then using the following instruction to start my client in python console:
`client = mph.Client(cores=8,port=2036) `

At the end of my script, when I'm trying to close it,  i'm using the instructions: 
```
client.disconnect()
del client
```

I can see on the comsol server cmd window that my client has been disconnecting from the server... all good

This being done, I have no possibility to start again the client with the first instruction. When doing so, I obtain the following:

> Only one client can be instantiated at a time.
> Traceback (most recent call last):
> 
>   Cell In[16], line 1
>     client = mph.Client(cores=8,port=2036) #
> 
>   File ~\anaconda3\envs\Last240416\Lib\site-packages\mph\client.py:151 in __init__
>     raise NotImplementedError(error)
> 
> NotImplementedError: Only one client can be instantiated at a time.

It seems that the client is still there... Strange...

I always have to kill the console to get the chance to get back to work.

What is the trick I'm missing there?

Thanks for your input!

With regards

Magaud

**Comment by john-hen** [comment]
Hi. There is no trick. Ending the Python session *is* the way to properly close the client.

This is no different from the GUI. You can't close the client without closing the GUI.

**Comment by Magaud59** [comment]
Thanks for your answer John-Hen.
I a former script, I was able to properly stop and start the client... but I was not using the GUI on the background. So... that's the reason right?
As soon as the GUI in "On", I'm kind of tricked.

All good, I'm fine with this... I just have to accept it ;)

I'm wishing you a good day.

With regards

M
